### PR TITLE
Major version updates

### PR DESCRIPTION
**Pull request by Shabaner**
Upgraded Kafka Version, IAM Authenticator version.
Changed Twitter API to Twitter Data Generator.
Upgrade Gradlew and Java to Version 21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,23 +12,111 @@ You can find a further details and a more thorough description and discussion of
...
+9. Deploy the cloud formation template `./misc/cfn-ecs-fargate.yml` in your account
+
+
+
+
+## Project Structure
+
+- `/kafka-streams-msk/*`: Kafka Streams implementation including Dockerfile
+- `/misc/`: Contains the cloud formation definition for the ECS and MSK cluster
+- `/twitter-stream-producer/`: Twitter Stream producer
```
Comment (awsalialem) [inline README.md:68]: Can we rename twitter to X everywhere?

**Review by awsalialem — COMMENTED** [review]
If possible rename all twitters in class names and readme files to X.
Are we removing the code to call X platform API which fetches posts and write in Kafka?

**Comment by Shabaner** [comment]
> If possible rename all twitters in class names and readme files to X. Are we removing the code to call X platform API which fetches posts and write in Kafka?

The Twitter API changed pricing plans to have only paid plans. The used API does now cost and needs a subscription.

File: README.md (hunk 1 above)
Comment (Shabaner) [reply README.md:68]: The name of twitter api still exists, therefore I suggest to leave it for the moment.

**Review by awsalialem — APPROVED** [review]
Looks good to me

**Review by awsalialem — APPROVED** [review]
Looks good to me

**Pull request merged**